ivy/functional/ivy/creation.py

```diff
@@ -69,6 +68,22 @@ def new_fn(*args, dtype=None, **kwargs):
         specifically for the backend implementations of asarray.
         """
         if isinstance(args[0], list):
+
+            # This is a really dirty hack to get around the fact that
+            # ivy.NativeArray is a Union object And repeating its
+            # definition here is going to drive someone mad when the
+            # real definition changes.
+
+            # TODO: When / if Ivy updates to Python3.10, change this
+            #  to pass a type union into isinstance and use the fact
+            #  that Union objects are "self-flattening" to make this
+            #  really elegant
+            if isinstance(args[0][0], ivy.Array) or type(args[0][0]).__name__ in str(
+                ivy.NativeArray
+            ):
+                return to_ivy(ivy.stack(to_native(args[0]), *(args[1:])))
```

Comment:
As written in the PR the code works just fine for me for 1D lists of arrays, I haven't tested it for higher dimension arrays though. But on my machine your Torch code will run and give the same result as the Tensorflow version would. I had `ivy.stack` instead of `fn` because in this wrapper `fn` will always be one of the `asarray` implementations but one of those crashes here, so idea was to use `ivy.stack` to add the behaviour to the Torch backend, but I hadn't thought about how it affects the other backends, I'll move the change over to the Torch backend and I'll get some tests running for different inputs.

Reply:
I've moved the fix into just the Torch backend, and I've edited the test for `asarray` to test lists of elements and lists of native array classes, I'm not sure if the changes to the test should stay but I've added them for now to ensure that this fix works, the tests pass for me on every backend.